### DISCUSSION TITLE
Trigger Hotkey and Text messages from a Javascript plugin?

### DISCUSSION THREAD

**Discussion by jamesmandrews** [elgatosf/elgato#40] [Help]
I play GTAV RP on NoPixel's public server.  As part of their development they have created a list of 500 animations players can use to create emotion in game.  I have 100+ of these I want to setup as buttons for the stream deck, which is a tedious process.  Currently I have a multi-action setup for a few.  The multi action is comprised of a hotkey to trigger the emote dialog box.  a delay, a text trigger to send the emote, another delay, and then another hotkey to trigger submission.  it works well.  What I would like to do is create a plugin that instead has this functionality in it, and a select box with a list of all the emote actions listed, so players can drag to button and select the emotion they want to link to the button.  However, I can't for the life of me figure out if I can call hotkey or text system calls from a Javascript based plugin using websockets and if I can, how I would do it.  Could someone please help, and maybe offer me a pathway to build this?

**Comment by ChekTek** [comment]
Hi @jamesmandrews 

Apologies that this went unanswered for so long! My recommendation would be to build a Node.js addon using something like C++ or Rust. This can be accomplished using the [N-API](https://nodejs.org/api/n-api.html), which allows you to call native code from javascript in a Node.js Stream Deck plugin.
